### PR TITLE
Add test to verify BUILD- and cmake-files are in sync with src/Makefile.am

### PR DESCRIPTION
**Pull request by Yannic**

### CODE REVIEW

File: build_files_updated_unittest.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# This script verifies that BUILD files and cmake files are in sync with src/Makefile.am
```
Comment (acozzette) [inline build_files_updated_unittest.sh]: I think you may need to add `set -e` at the top of this test script so that a failing command causes the script to return a non-zero error code. Otherwise one of the `diff` commands could detect a difference without the test failing.

File: cmake/extract_includes.bat.in [hunk 2]
```diff
@@ -56,7 +56,6 @@ copy "${PROTOBUF_SOURCE_WIN32_PATH}\..\src\google\protobuf\implicit_weak_message
 copy "${PROTOBUF_SOURCE_WIN32_PATH}\..\src\google\protobuf\inlined_string_field.h" include\google\protobuf\inlined_string_field.h
 copy "${PROTOBUF_SOURCE_WIN32_PATH}\..\src\google\protobuf\io\coded_stream.h" include\google\protobuf\io\coded_stream.h
 copy "${PROTOBUF_SOURCE_WIN32_PATH}\..\src\google\protobuf\io\gzip_stream.h" include\google\protobuf\io\gzip_stream.h
-copy "${PROTOBUF_SOURCE_WIN32_PATH}\..\src\google\protobuf\io\io_win32.h" include\google\protobuf\io\io_win32.h
```
Comment (acozzette) [inline cmake/extract_includes.bat.in]: We need to keep this header here because it's needed on Windows. I think if you edit `src/Makefile.am` to move it out of `libprotobuf_lite_la_SOURCES` and into `nobase_include_HEADERS` then the `update_file_lists.sh` script will keep it here.

**Comment by acozzette** [comment]
@Yannic Thank you for the pull request. I made a couple small comments but this looks great and should make it much easier to keep our different build systems in sync.

File: build_files_updated_unittest.sh (hunk 1 above)
Comment (Yannic) [reply build_files_updated_unittest.sh]: Done!

**Comment by acozzette** [comment]
@Yannic It looks like your new test is failing in the CI run, do you know what that is about?

**Comment by Yannic** [comment]
@acozzette No idea, could be a platform thing. It passes when executed locally. Added a simple script to extract Bazel test logs from kokoro. Can we run the tests again so we can see what's happening?

**Comment by Yannic** [comment]
@acozzette The issue was that the repo is mounted as read-only. This is fixed now, ptal.
